### PR TITLE
Add project dependencies

### PR DESCRIPTION
**Pull request by accua**
- Adds HTTP client httpoison
- JSON lib poison
- ex_doc for doc generation
- mix_test_watch runs tests each time you save, good for TDD.

### CODE REVIEW

**Review by ckrailo — APPROVED** [review]

File: mix.exs [hunk 1]
```diff
@@ -21,8 +21,10 @@ defmodule Politex.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      # {:dep_from_hexpm, "~> 0.3.0"},
-      # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"},
+      {:httpoison, "~> 1.3.0"},
```
Comment (doughsay) [inline mix.exs:24]: Have you considered other HTTP clients?  httpoison is definitely the most popular, but it's architected more like httparty from Ruby-land.  If you want something more like Faraday from Ruby-land, consider checking out [tesla](https://github.com/teamon/tesla).  I've been using it in production and like the middleware architecture it uses.

File: mix.exs [hunk 2]
```diff
@@ -21,8 +21,10 @@ defmodule Politex.MixProject do
...
+      {:poison, "~> 4.0.1"},
```
Comment (doughsay) [inline mix.exs:25]: The community is largely moving to [jason](https://github.com/michalmuskala/jason).  It's faster than poison, and many projects have jumped ship already.  I'd recommend using it from the start.

Comment (bzitzow) [reply mix.exs:25]: 👍

File: mix.exs (hunk 1 above)
Comment (bzitzow) [reply mix.exs:24]: 👍

Comment (ckrailo) [reply mix.exs:24]: Great recommendations, thanks! These were originally chosen from off-the-cuff recommendations during a lunch chat with McCord, but we're not opposed to using different stuff. ¯\\\_(ツ)\_/¯

**Comment by ckrailo** [comment]
Merging this! We should look into Tesla and either switch or at least make a branch that implements Tesla just to work with it.

**Pull request merged**